cql3/statements/schema_altering_statement.cc

```diff
@@ -83,6 +86,18 @@ schema_altering_statement::execute(query_processor& qp, service::query_state& st
     co_return std::move(result);
 }
 
+future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>> schema_altering_statement::prepare_schema_mutations(query_processor& qp, const query_options& options, api::timestamp_type) const {
+    // derived class must implement one of prepare_schema_mutations overloads
+    on_internal_error(logger, "not implemented");
+    co_return std::make_tuple(::shared_ptr<cql_transport::event::schema_change>(nullptr), std::vector<mutation>{}, cql3::cql_warnings_vec{});
+}
+
+future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, cql3::cql_warnings_vec>> schema_altering_statement::prepare_schema_mutations(query_processor& qp, service::query_state& state, const query_options& options, service::mutations_collector& mc) const {
+    auto [ret, muts, cql_warnings] = co_await prepare_schema_mutations(qp, options, mc.write_timestamp());
+    mc.add_mutations(std::move(muts));
+    co_return std::make_tuple(ret, cql_warnings);
+}
```

Comment:
By the way, regardless of whether you choose to modify all implementors of prepare_schema_mutations to take a mutations collector or you keep the current split, this looks like a good example of a change that could be moved to a separate commit.

Reply:
> IMO it would be better to convert all implementors of prepare_schema_mutations

sadly it's 20 of them. but I can do it as I want to push for more usage of mutation_collector.


If all reviewers are fine with putting mutation_collector::unused() in all those places, and some additional `if` in prepare_schema_mutations when calling `mc.add_mutations` to support not calling it?

Reply:
> If all reviewers are fine with putting mutation_collector::unused() in all those places, and some additional `if` in `prepare_schema_mutations` when calling `mc.add_mutations` to support not calling it?

Aren't all schema altering statements supposed to use a group0 guard? You can pass a proper mutation collector with a guard in it in all cases, no?

Reply:
@piodul are we good with resolving this conversation or you have further comments?